src/stable/lib/stable_structures/stable_json.ts

```diff
@@ -198,6 +232,38 @@ export function jsonReviver(_key: string, value: any): any {
         if (typeof value.__biguint64array__ === 'object') {
             return BigUint64Array.from(value.__biguint64array__);
         }
+
+        if (typeof value.__float32array__ === 'object') {
+            return Float32Array.from(value.__float32array__);
+        }
+
+        if (typeof value.__float64array__ === 'object') {
+            return Float64Array.from(value.__float64array__);
+        }
+
+        if (Array.isArray(value.__map__)) {
```

Comment:
Change other checks to `isArray` as appropriate

Reply:
done